charlib/characterizer/cell.py

```
@@ -139,7 +139,35 @@ def match_direction(pin_name):
         ## 4. Add as much liberty data as we can right now
         self.liberty = liberty.Group('cell', name)
         self.liberty.add_attribute('area', cell_config.get('area', 0.0), 2)
-        for pin in self.all_pins():
+        def _to_lib_expr(pin):
```

Ruff (0.15.17)

[warning] Missing return type annotation for private function `_to_lib_expr`

(ANN202)